src/DocumentFormat.OpenXml/SimpleTypes/OpenXmlComparableReferenceType.cs

```diff
@@ -0,0 +1,179 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+
+namespace DocumentFormat.OpenXml
+{
+    /// <summary>
+    /// Represents a comparable and equatable reference.
+    /// </summary>
+    /// <typeparam name="T">The type of the value.</typeparam>
+    public abstract class OpenXmlComparableReferenceType<T> : OpenXmlSimpleType,
+        IComparable, IComparable<OpenXmlComparableReferenceType<T>>, IEquatable<OpenXmlComparableReferenceType<T>>
+        where T : class, IComparable, IComparable<T>, IEquatable<T>
+    {
+        /// <summary>
+        /// Creates a new instance of <see cref="OpenXmlComparableReferenceType{T}"/>.
+        /// </summary>
+        private protected OpenXmlComparableReferenceType()
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="OpenXmlComparableReferenceType{T}"/>
+        /// class by deep copying the supplied <see cref="OpenXmlComparableReferenceType{T}"/>
+        /// value.
+        /// </summary>
+        /// <param name="source">The source <see cref="OpenXmlComparableReferenceType{T}"/> instance.</param>
+        private protected OpenXmlComparableReferenceType(OpenXmlComparableReferenceType<T> source) : base(source)
+        {
+        }
+
+        /// <summary>
+        /// Gets or sets the value of this instance.
+        /// </summary>
+        public abstract T Value { get; set; }
```

Comment:
Does this need to be abstract? Or can it just be virtual (or even non-virtual)? It would nice to follow the pattern of `OpenXmlSimpleValue<T>`

Reply:
I thought about that as well. Currently, only three subclasses inherit from `OpenXmlComparableReferenceType<T>`:

- `Base64BinaryValue`,
- `HexBinaryValue`, and
- `StringValue`.

All of those basically delegate to `OpenXmlSimpleType.TextValue`. Therefore, I thought about two options:

1. turn `OpenXmlComparableReferenceType<T>` into something like `OpenXmlStringType` and provide a concrete implementation of the `Value` property, or
2. leave `OpenXmlComparableReferenceType<T>` generic to enable reuse for non-string types.

For the second option, which I chose for this implementation, I thought it was more elegant to just keep the that property abstract. However, we could also go for Option 1.

Reply:
Ah - I hadn't realized it was a generic type. I like the current implementation better so it's more reusable.